### PR TITLE
Extended tslint.json from tslint:all

### LINKED ISSUE #608: Extend tslint.json from tslint:all
This will require disabling a bunch of rules, many of which will be listed in #491. Our `tslint.json` predates the ability to extend configurations.

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
Changes `getRuleNames` utility to `getContribuRuleNames` and no longer retrieve core TSLint rules. Also adds comments and sections in our `tslint.json` to help distinguish classes of rule settings.

Fixes #608.

### CODE REVIEW

File: build-tasks/validate-config.js [hunk 1]
```diff
@@ -2,62 +2,35 @@
...
 
 const errors = [];
-getAllRuleNames().forEach(ruleName => {
-    if (rulesToSkip.has(ruleName)) {
+getContribRuleNames().forEach(ruleName => {
+    if (disabledRules.has(ruleName)) {
         return;
     }
+
     if (tslintConfig.rules[ruleName] !== true && tslintConfig.rules[ruleName] !== false) {
```
Comment (IllusionMH) [inline build-tasks/validate-config.js]: There is still a problem with this logic.
Looks like it won't allow to do `"rule": [false, "param"]` but still pass on `"rule": false`.
Maybe it makes sense to update logic with whole rule?

Comment (JoshuaKGoldberg) [reply build-tasks/validate-config.js]: Agreed, this is a good time.

**Comment by IllusionMH** [comment]
👍

File: build-tasks/validate-config.js (hunk 1 above)
Comment (JoshuaKGoldberg) [reply build-tasks/validate-config.js]: @IllusionMH does this look good to you?

Comment (IllusionMH) [reply build-tasks/validate-config.js]: Yes, this should cover all necessary cases for this repo (no rules with different severity).

**Pull request merged**